pkg/cmd/admin/validate/node.go

```diff
@@ -0,0 +1,98 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	"github.com/openshift/origin/pkg/cmd/server/api/validation"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	ValidateNodeConfigRecommendedName = "node-config"
+
+	validateNodeConfigLong = `
+Validate the configuration file for a node.
+
+This command validates that a configuration file intended to be used for a node is 
+created correctly. 
+`
+
+	valiateNodeConfigExample = ` // Validate node configuration file
+  $ %s openshift.local.config/master/node-config.yaml`
+)
+
+type ValidateNodeConfigOptions struct {
+	// NodeConfigFile is the location of the config file to be validated
+	NodeConfigFile string
+
+	// Stderr is the writer to write warnings to
+	Stderr io.Writer
+
+	// Out is the writer to write output to
+	Out io.Writer
+}
+
+func NewValidateNodeConfigOptions() *ValidateNodeConfigOptions {
+	return &ValidateNodeConfigOptions{
+		Stderr: os.Stderr,
+	}
+}
+
+// NewCommandValidateMasterConfig provides a CLI handler for the `validate all-in-one` command
+func NewCommandValidateNodeConfig(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := NewValidateNodeConfigOptions()
+	options.Out = out
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SOURCE", name),
+		Short:   "Validate the configuration file for a node",
+		Long:    validateNodeConfigLong,
+		Example: fmt.Sprintf(valiateNodeConfigExample, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(args, f); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := options.Validate(); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			options.Out.Write([]byte(fmt.Sprintf("SUCCESS: Validation succeded for file: %s", options.NodeConfigFile)))
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *ValidateNodeConfigOptions) Complete(args []string, f *clientcmd.Factory) error {
+	switch len(args) {
+	case 0:
+		return errors.New("the path to the file being validated is required")
+	case 1:
+		o.NodeConfigFile = args[0]
+		return nil
+	default:
+		return errors.New("at most one file can be identified for validation")
+	}
+}
+
+func (o *ValidateNodeConfigOptions) Validate() error {
+	nodeConfig, err := configapilatest.ReadAndResolveNodeConfig(o.NodeConfigFile)
+	if err != nil {
+		return err
+	}
+
+	results := validation.ValidateNodeConfig(nodeConfig)
```

Comment:
> I feel like the best way to do this would be to add a DescribeValidationError method to pkg/cmd/cli/describe/. Is this a good idea?

I don't think so.  That's supposed to be for API objects.